src/Illuminate/Http/TransformableResponse.php

```diff
@@ -0,0 +1,333 @@
+    {
+        $pattern = '/'.str_replace('.*.', '\.([0-9])+\.', $rule).'/';
+
+        return array_reduce($gatheredRules,
+            function ($validRules, $rule) use ($pattern, $valueForValidOnes) {
+                preg_match($pattern, $rule, $matches);
+
+                if ($matches) {
+                    $validRules[$matches[0]] = $valueForValidOnes;
+                }
+
+                return $validRules;
+            },
+        []);
+    }
+
+    /**
+     * Apply visbility rules to given data.
+     *
+     * @param  array $rules
+     * @return array
+     */
+    protected function applyVisibilityRules(array $data, array $rules)
+    {
+        if (empty($rules)) {
+            return $data;
+        }
+
+        $data = $this->showFields($data, $rules);
+        $data = $this->hideFields($data, $rules);
+
+        return $data;
+    }
+
+    /**
+     * Apply rules over the fields that must be displayed.
+     *
+     * @param  array $data
+     * @param  array $rules
+     * @return array
+     */
+    protected function showFields(array $data, array $rules)
+    {
+        $applicableRules = array_filter($rules);
+
+        if (empty($applicableRules)) {
+            return $data;
+        }
+
+        return array_reduce(array_keys($applicableRules),
+            function ($transformedData, $attribute) use ($data) {
+                if ($value = Arr::get($data, $attribute)) {
+                    Arr::set(
+                        $transformedData,
+                        $this->applyRenamingRules($attribute),
+                        $this->applyMutationRules(
+                            $attribute,
+                            $this->applyCastingRules($attribute, $value)
+                        )
+                    );
+                }
+
+                return $transformedData;
+            },
+        []);
+    }
+
+    /**
+     * Apply rules over the fields that must be hidden.
+     *
+     * @param  array $data
+     * @param  array $rules
+     * @return array
+     */
+    protected function hideFields(array $data, array $rules)
+    {
+        $applicableRules = array_filter($rules, function ($rule) {
+            return ! $rule;
+        });
+
+        if (empty($applicableRules)) {
+            return $data;
+        }
+
+        return array_reduce(array_keys($applicableRules),
+            function ($transformedData, $rule) {
+                if (Arr::has($transformedData, $rule)) {
+                    Arr::forget($transformedData, $rule);
+                }
+
+                return $transformedData;
+            },
+        $data);
+    }
+
+    /**
+     * Apply casting rules to given value.
+     *
+     * @param  string $attribute
+     * @param  mixed $value
+     * @return mixed
+     */
+    protected function applyCastingRules($attribute, $value)
+    {
+        switch ($this->getCastType($attribute)) {
+            case 'int':
+            case 'integer':
```

Comment:
Yep, thats something I wanted to talk here.

I didn't want to refactor Eloquent to use the switch that it uses to define the casting. What should I do here ?